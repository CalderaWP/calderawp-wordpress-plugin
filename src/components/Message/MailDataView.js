import React from "react";
function createMarkup(message) {
	return {__html: message};
}

function MyComponent() {
	return <div dangerouslySetInnerHTML={createMarkup()} />;
}

export const MailDataView = ({mailData}) => (

	<div className={'wp-block-columns has-2-columns'}>
		<div className={'wp-block-column'}>
			<h3>Headers</h3>
			<ul>
				{Object.keys(mailData).map(key => {
					if (mailData.hasOwnProperty(key) && ![
						'message',
						'headers',
						'attachments'
					].includes(key)) {
						const datum = mailData[key];
						let display = datum;
						if( 'html' === key ){
							display = datum ? 'HTML' : 'Plain text';
						}
						if (datum) {
							return (<li key={key}><span>{key.charAt(0).toUpperCase() + key.slice(1)}:</span> {display}</li>);
						}
					}

				})}
			</ul>
		</div>
		<div className={'wp-block-column'}>
			<h3>Message</h3>
			<div dangerouslySetInnerHTML={createMarkup(mailData.message)} />
		</div>
	</div>


)
