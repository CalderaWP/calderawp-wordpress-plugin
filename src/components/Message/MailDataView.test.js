import React from "react";
import renderer from "react-test-renderer";
import "@babel/polyfill";
import {MailDataView} from "./MailDataView";

const mailData = {
	attachments: [],

	bcc: false,

	csv: false,
	from: "test@test.com",
	from_name: "Caldera Forms Notification",
	html: true,
	message: "<p><strong></strong></p>\n<div style=\"margin-bottom:20px;\">Mike</div>\n<p><strong></strong></p>\n<div style=\"margin-bottom:20px;\">Pollock</div>\n<p><strong></strong></p>\n<div style=\"margin-bottom:20px;\">jâ€¦",
	recipients:
		['roy@hiroy.club','alsoroy@hiroy.club'],
	replyto: "test@test.com",
	subject: "Contact Form 1"
};
describe( 'MailDataView' , () => {
	it( 'matches snapshot', () => {
		expect(renderer.create(<MailDataView mailData={mailData}/>).toJSON()
		).toMatchSnapshot()
	})
});
