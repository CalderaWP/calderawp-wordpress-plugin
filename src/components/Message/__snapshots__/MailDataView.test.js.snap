// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailDataView matches snapshot 1`] = `
<div
  className="wp-block-columns has-2-columns"
>
  <div
    className="wp-block-column"
  >
    <h3>
      Headers
    </h3>
    <ul>
      <li>
        <span>
          From
          :
        </span>
         
        test@test.com
      </li>
      <li>
        <span>
          From_name
          :
        </span>
         
        Caldera Forms Notification
      </li>
      <li>
        <span>
          Html
          :
        </span>
         
        HTML
      </li>
      <li>
        <span>
          Recipients
          :
        </span>
         
        roy@hiroy.club
        alsoroy@hiroy.club
      </li>
      <li>
        <span>
          Replyto
          :
        </span>
         
        test@test.com
      </li>
      <li>
        <span>
          Subject
          :
        </span>
         
        Contact Form 1
      </li>
    </ul>
  </div>
  <div
    className="wp-block-column"
  >
    <h3>
      Message
    </h3>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p><strong></strong></p>
        <div style=\\"margin-bottom:20px;\\">Mike</div>
        <p><strong></strong></p>
        <div style=\\"margin-bottom:20px;\\">Pollock</div>
        <p><strong></strong></p>
        <div style=\\"margin-bottom:20px;\\">jâ€¦",
        }
      }
    />
  </div>
</div>
`;
